### PR TITLE
Add indexer connector testtool enhancement

### LINKED ISSUE #18881: Enhance testtool capacity for indexer connector
### Description

This issue focuses on enhancing the capacity of the TestTool within the Indexer Connector. The TestTool is a critical component for testing and validating the functionality of the component.

### Tasks

1. Load Template into OpenSearch:
   - Extend the TestTool to support loading a template into OpenSearch using two flags: one for the configuration path and another for the template path.
   - Command: `testtool -c config.json -t template.json`

2. Handle Template Mapping Errors:
   - Implement error handling to respond when a template cannot be mapped correctly in OpenSearch.

3. Send Events from JSON File:
   - Enhance the TestTool to send events from a JSON file to validate data integration.
   - Command: `testtool -c config.json -t template.json -e events.json`

4. Send Autogenerated Events:
   - Extend the TestTool to generate and send events automatically to simulate real-world scenarios.
   - Command: `testtool -c config.json -t template.json -a true -n 5000` (send 5000 autogenerated events)

### Definition of Done

- The TestTool is updated to load templates into OpenSearch using the provided configuration and template paths.
- Error handling is implemented to handle template mapping errors.
- The TestTool can effectively send events from a JSON file to validate data integration.
- Autogenerated events can be generated and sent by the TestTool as specified.
- Code changes are reviewed, tested, and merged into the main codebase.
- Documentation is updated to reflect the changes made.
- Quality assurance checks are conducted to validate the enhanced TestTool's functionality.

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #18881 |

### Description

The purpose of this PR is to add the capacity to test the indexer connector component with a testtool.

### CODE REVIEW

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp [hunk 1]
```diff
@@ -12,50 +12,74 @@
...
+    auto selector = std::make_shared<ServerSelector>(config.at("indexer").at("hosts"));
+
+    // Read template file.
+    std::ifstream templateFile(templatePath);
+    if (!templateFile.is_open())
+    {
+        throw std::runtime_error("Could not open template file.");
+    }
+
+    nlohmann::json templateData;
```
Comment (pereyra-m) [inline src/shared_modules/indexer_connector/src/indexerConnector.cpp:35]: Coding style: the `nlohmann` variable can be initialized directly from the `std::ifstream` object.

File: src/shared_modules/indexer_connector/testtool/main.cpp [hunk 2]
```diff
@@ -1,40 +1,152 @@
...
+            const auto n = cmdArgParser.getNumberOfEvents();
+            // Read template file.
+
+            std::ifstream templateFile(cmdArgParser.getTemplateFilePath());
+            if (!templateFile.is_open())
+            {
+                throw std::runtime_error("Could not open template file.");
+            }
+
+            nlohmann::json templateData;
```
Comment (pereyra-m) [inline src/shared_modules/indexer_connector/testtool/main.cpp:123]: Coding style: the `nlohmann` variable can be initialized directly from the `std::ifstream` object.

File: src/shared_modules/indexer_connector/testtool/main.cpp [hunk 3]
```diff
@@ -1,40 +1,152 @@
...
+        // If the events file path is empty, then the events are generated
+        // automatically.
+        if (!cmdArgParser.getEventsFilePath().empty())
+        {
+            std::ifstream eventsFile(cmdArgParser.getEventsFilePath());
+            if (!eventsFile.is_open())
+            {
+                throw std::runtime_error("Could not open events file.");
+            }
+            nlohmann::json events;
```
Comment (pereyra-m) [inline src/shared_modules/indexer_connector/testtool/main.cpp:106]: Coding style: the `nlohmann` variable can be initialized directly from the `std::ifstream` object.

**Review by pereyra-m — COMMENTED** [review]
Gj!

Just a coding style detail

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp [hunk 4]
```diff
@@ -12,50 +12,74 @@
...
 std::unordered_map<IndexerConnector*, std::unique_ptr<ThreadDispatchQueue>> QUEUE_MAP;
 constexpr auto DATABASE_WORKERS = 1;
+constexpr auto DATABASE_BASE_PATH = "queue/indexer/";
 
-IndexerConnector::IndexerConnector(const nlohmann::json& config)
+IndexerConnector::IndexerConnector(const nlohmann::json& config, const std::string& templatePath)
 {
     // Initialize publisher.
-    auto selector = std::make_shared<ServerSelector>(config.at("servers"));
+    auto selector = std::make_shared<ServerSelector>(config.at("indexer").at("hosts"));
```
Comment (GabrielEValenzuela) [inline src/shared_modules/indexer_connector/src/indexerConnector.cpp:26]: This allocation memory could be done after checking if the template file has been opened correctly.

**Review by GabrielEValenzuela — COMMENTED** [review]
GJ!

**Review by pereyra-m — APPROVED** [review]
LGTM!

**Pull request merged**